decimal.go

```diff
@@ -423,14 +454,37 @@ func (d Decimal) Copy() Decimal {
 	}
 }
 
+// RoundMath scales decimal value to an integer value with given exponent. On
+// exponent scale-down decimal value precision is preserved, on exponent
+// scale-up rounding with the given rounding rule is performed.
+func (d Decimal) RoundMath(exp int, rule RoundRule) Decimal {
```

Comment:
Naming of this function is confusing now since you can specify the rounding rule you want. I think naming this function `Round` and renaming existing `Round` function to `RoundMath` would make more sense.